pkg/management/postgres/utils/version.go

```diff
@@ -38,25 +38,30 @@ func GetPgVersion(db *sql.DB) (*semver.Version, error) {
 }
 
 func parseVersionNum(versionNum string) (*semver.Version, error) {
-	versionInt, err := strconv.Atoi(versionNum)
+	versionInt, err := strconv.ParseUint(versionNum, 10, 64)
 	if err != nil {
 		return nil, err
 	}
 
 	return &semver.Version{
-		Major: uint64(versionInt / 10000),       //nolint:gosec
-		Minor: uint64((versionInt / 100) % 100), //nolint:gosec
-		Patch: uint64(versionInt % 100),         //nolint:gosec
+		Major: versionInt / 10000,
+		Minor: (versionInt / 100) % 100,
+		Patch: versionInt % 100,
 	}, nil
 }
 
-// GetMajorVersion read the PG_VERSION file in the data directory
+// GetPgdataVersion read the PG_VERSION file in the data directory
 // returning the major version of the database
-func GetMajorVersion(pgData string) (int, error) {
+func GetPgdataVersion(pgData string) (semver.Version, error) {
```

Comment:
I'm not sure I agree with the new signature.
The PG_VERSION contrains only the major number, nothing else.
I don't think this is the right place to return a semver.Version

Reply:
The goal is to eliminate confusion and assign a clear, specific type to the value. We've already made similar adjustments for the int/uint version in the past, with consensus that using `semver` helps prevent misinterpretation when the source of the values isn't immediately apparent.